breakpoint.c

```diff
@@ -0,0 +1,60 @@
+#include "breakpoint.h"
+#include <assert.h>
+
+static inline int cmp(const void *arg0, const void *arg1)
+{
+    riscv_word_t *a = (riscv_word_t *) arg0, *b = (riscv_word_t *) arg1;
+    return (*a < *b) ? _CMP_LESS : (*a > *b) ? _CMP_GREATER : _CMP_EQUAL;
+}
+
+breakpoint_map_t breakpoint_map_init()
```

Comment:
Here, you shall name it `breakpoint_map_new` instead of `breakpoint_map_init. Quote from [Constructors in Python (__init vs __new__)](https://dev.to/pila/constructors-in-python-init-vs-new-2f9j)
> `__new__` is the first step of instance creation. It's called first and is responsible for returning a new instance of your class.
> In contrast, `__init__` doesn't return anything; it's only responsible for initializing the instance after it's been created. 

The above was dedicated to Python, but name scheme is still making sense to C APIs.